spec/controllers/assignments_controller_spec.rb

```diff
@@ -174,8 +175,9 @@
       context 'when the end date is before the start date' do
         let(:end_date) { Date.yesterday.to_fs(:db) }
 
-        it 'redirects back' do
-          expect { submit }.to redirect_back
+        it 'stays on the the generator page' do
+          submit
+          expect(response).to render_template :rotation_generator
```

Comment:
```suggestion
          expect(response).to have_http_status(:unprocessable_entity)
```
Another case of "this is how things used to be done with controller actions but in modern day we keep things focused on the HTTP request/response itself"